tests/acceptance/core/service/factory/cluster.go

```diff
@@ -0,0 +1,62 @@
+package factory
+
+import (
+	"fmt"
+	"path/filepath"
+	"strconv"
+
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	"github.com/k3s-io/k3s/tests/acceptance/shared/util"
+
+	. "github.com/onsi/ginkgo/v2"
+)
+
+func BuildCluster(g GinkgoTInterface, destroy bool) (string, error) {
+	basepath := util.GetBasepath()
+	tfDir, err := filepath.Abs(basepath + "/modules/k3scluster")
+	if err != nil {
+		return "", err
+	}
+	varDir, err := filepath.Abs(basepath + "/modules/k3scluster/config/local.tfvars")
+	if err != nil {
+		return "", err
+	}
+
+	terraformOptions := terraform.Options{
+		TerraformDir: tfDir,
+		VarFiles:     []string{varDir},
+	}
+
+	util.NumServers, err = strconv.Atoi(terraform.GetVariableAsStringFromVarFile(g, varDir,
+		"no_of_server_nodes"))
+	if err != nil {
+		return "", err
+	}
+
+	util.NumAgents, err = strconv.Atoi(terraform.GetVariableAsStringFromVarFile(g, varDir,
+		"no_of_worker_nodes"))
+	if err != nil {
+		return "", err
+	}
+
+	util.ClusterType = terraform.GetVariableAsStringFromVarFile(g, varDir, "cluster_type")
+	util.ExternalDb = terraform.GetVariableAsStringFromVarFile(g, varDir, "external_db")
+	util.AwsUser = terraform.GetVariableAsStringFromVarFile(g, varDir, "aws_user")
+	util.AccessKey = terraform.GetVariableAsStringFromVarFile(g, varDir, "access_key")
+
+	if destroy {
+		fmt.Printf("Cluster is being deleted")
+		terraform.Destroy(g, &terraformOptions)
+		return "cluster destroyed", err
+	}
+
+	fmt.Printf("Creating Cluster")
+
+	terraform.InitAndApply(g, &terraformOptions)
+	util.KubeConfigFile = "/tmp/" + terraform.Output(g, &terraformOptions, "kubeconfig") + "_kubeconfig"
+	util.ServerIPs = terraform.Output(g, &terraformOptions, "master_ips")
+	util.AgentIPs = terraform.Output(g, &terraformOptions, "worker_ips")
+	util.RenderedTemplate = terraform.Output(g, &terraformOptions, "rendered_template")
+
+	return "cluster created", err
```

Comment:
```suggestion

    cluster := util.Cluster{}
	terraform.InitAndApply(g, &terraformOptions)
    cluster.KubeConfigFile = "/tmp/" + terraform.Output(g, &terraformOptions, "kubeconfig") + "_kubeconfig"
	cluster.ServerIPs = terraform.Output(g, &terraformOptions, "master_ips")
	cluster.AgentIPs = terraform.Output(g, &terraformOptions, "worker_ips")
	cluster.RenderedTemplate = terraform.Output(g, &terraformOptions, "rendered_template")

	return cluster, err
```

Reply:
Still not resolved

Reply:
this one as per comment above i will not apply these changes.

Reply:
the reason is here: 
to do that i would need to instanciate this struct or have a build function for that and everywhere that i need those values i would need to create that dependency in a lot of files

So here i really think the best way to go is creating global vars and actually makes more sense because they are global configurations values